onnxruntime/core/framework/session_state.h

```diff
@@ -129,7 +130,12 @@ class SessionState {
   AllocatorPtr GetAllocator(const OrtMemoryInfo& location) const noexcept;
 
   /** Get the allocator for a given OrtDevice. The first allocator that matches will be returned. */
-  AllocatorPtr GetAllocator(OrtDevice device) const noexcept;
+  AllocatorPtr GetAllocator(const OrtDevice& device) const noexcept;
+
+  /*
+  * Get allocators. CANNOT be const member function as allocators_ will be changed after SessionState's initialization for shared allocator scenario (InferenceSession::UpdateSessionStateAllocatorsWithSharedAllocators())
+  */
+  std::map<OrtDevice, AllocatorPtr>& GetAllocators() { return *allocators_; }
```

Comment:
Why return by ref? Did you mean const-ref?

Reply:
returned by ref is because there should be one instance of allocators map object when it is referenced among infernece_session, session_state and kernel_info.

The reason I didn't make it const is because, as I explained in the comment, the allocators_ object cannot be const as it will be changed to merge shared allocators in InferenceSession::UpdateSessionStateAllocatorsWithSharedAllocators() after SessionState's initialization

Reply:
add const decorator